tests/parametric/test_config_consistency.py

```diff
@@ -43,3 +44,39 @@ def test_tracing_disabled(self, library_env, test_agent, test_library):
         assert (
             True
         ), "wait_for_num_traces raises an exception after waiting for 1 trace."  # wait_for_num_traces will throw an error if not received within 2 sec, so we expect to see an exception
+
+
+@scenarios.parametric
+@features.tracing_configuration_consistency
+class Test_Config_UnifiedServiceTagging:
+    @parametrize("library_env", [{}])
+    def test_default_version(self, library_env, test_agent, test_library):
+        with test_library:
+            with test_library.start_span(name="s1") as s1:
+                pass
+
+        traces = test_agent.wait_for_num_traces(1)
+        assert len(traces) == 1
+
+        span = find_span_in_traces(traces, s1.trace_id, s1.span_id)
+        assert span["service"] != "version_test"
+        assert "version" not in span["meta"]
+
+    @parametrize("library_env", [{"DD_SERVICE": "version_test", "DD_VERSION": "5.2.0"}])
+    def test_specific_version(self, library_env, test_agent, test_library):
+        with test_library:
+            with test_library.start_span(name="s1") as s1:
+                pass
+            with test_library.start_span(name="s2", service="no dd_service") as s2:
+                pass
+
+        traces = test_agent.wait_for_num_traces(2)
+        assert len(traces) == 2
+
+        span1 = find_span_in_traces(traces, s1.trace_id, s1.span_id)
+        assert span1["service"] == "version_test"
+        assert span1["meta"]["version"] == "5.2.0"
+
+        span2 = find_span_in_traces(traces, s2.trace_id, s2.span_id)
+        assert span2["service"] != "version_test"
```

Comment:
nit:
```suggestion
        assert span2["service"] != "no dd_service"
```

Reply:
If anything, shouldn't this be checking `assert span2["service"] == "no dd_service"`

Reply:
Oops, yes you're right! Let me fix it